chronos/internal/asyncengine.nim

```diff
@@ -10,6 +10,94 @@
 
 {.push raises: [].}
 
+## This module implements the core asynchronous engine / dispatcher.
+##
+## The dispatcher acts as a kind of loop that wires events coming from the
+## operating system to tasks scheduled by the program. You must call
+## ``poll`` on it (or a function which does so for you such as ``waitFor`` or
+## ``runForever``) in order to poll for any outstanding events.
+##
+## The underlying implementation is based on epoll on Linux, IO Completion Ports
+## on Windows and select on other operating systems.
+##
+## The ``poll`` function will not, on its own, return any events. Instead
+## an appropriate ``Future`` object will be completed. A ``Future`` is a
+## type which holds a value which is not yet available, but which *may* be
+## available in the future. You can check whether a future is finished
+## by using the ``finished`` function. When a future is finished it means that
+## either the value that it holds is now available or it holds an error instead.
+## The latter situation occurs when the operation to complete a future fails
+## with an exception. You can distinguish between the two situations with the
+## ``failed`` function.
+##
+## Future objects can also store callback procedures which will be called
+## automatically once the future completes.
+##
+## Futures therefore can be thought of as an implementation of the proactor
+## pattern. In this
+## pattern you make a request for an action, and once that action is fulfilled
+## a future is completed with the result of that action. Requests can be
+## made by calling the appropriate functions. For example: calling the ``recv``
+## function will create a request for some data to be read from a socket. The
```

Comment:
This is confusing because behavior is different for different OS, what you describing here is more about Windows approach when you creating request and waiting for data, in Linux/BSD you are not creating request, you first waiting for signal that channel you want to read has some data...

Reply:
81a6584a44c69c03c47117c848dbbe63fac79fb5